# Анализ предложенных датасетов

## Описание всех датасетов с Kaggle:
### 1. IBM HR Analytics Employee Attrition & Performance

Uncover the factors that lead to employee attrition and explore important questions such as ‘show me a breakdown of distance from home by job role and attrition’ or ‘compare average monthly income by education and attrition’. This is a fictional data set created by IBM data scientists.

#### Education
##### 1 'Below College'
##### 2 'College'
##### 3 'Bachelor'
##### 4 'Master'
##### 5 'Doctor'

#### EnvironmentSatisfaction
##### 1 'Low'
##### 2 'Medium'
##### 3 'High'
##### 4 'Very High'

#### JobInvolvement
##### 1 'Low'
##### 2 'Medium'
##### 3 'High'
##### 4 'Very High'

#### JobSatisfaction
##### 1 'Low'
##### 2 'Medium'
##### 3 'High'
##### 4 'Very High'

#### PerformanceRating
##### 1 'Low'
##### 2 'Good'
##### 3 'Excellent'
##### 4 'Outstanding'

#### RelationshipSatisfaction
##### 1 'Low'
##### 2 'Medium'
##### 3 'High'
##### 4 'Very High'

#### WorkLifeBalance
##### 1 'Bad'
##### 2 'Good'
##### 3 'Better'
##### 4 'Best'

### 2. Employee Attrition Classification Dataset
The Synthetic Employee Attrition Dataset is a simulated dataset designed for the analysis and prediction of employee attrition. It contains detailed information about various aspects of an employee's profile, including demographics, job-related features, and personal circumstances.

The dataset comprises 74,498 samples, split into training and testing sets to facilitate model development and evaluation. Each record includes a unique Employee ID and features that influence employee attrition. The goal is to understand the factors contributing to attrition and develop predictive models to identify at-risk employees.

This dataset is ideal for HR analytics, machine learning model development, and demonstrating advanced data analysis techniques. It provides a comprehensive and realistic view of the factors affecting employee retention, making it a valuable resource for researchers and practitioners in the field of human resources and organizational development.

#### FEATURES:

##### Employee ID: A unique identifier assigned to each employee.#
##### Age: The age of the employee, ranging from 18 to 60 years.
##### Gender: The gender of the employee
##### Years at Company: The number of years the employee has been working at the company.
##### Monthly Income: The monthly salary of the employee, in dollars.
##### Job Role: The department or role the employee works in, encoded into categories such as Finance, Healthcare, Technology, Education, and Media.
##### Work-Life Balance: The employee's perceived balance between work and personal life, (Poor, Below Average, Good, Excellent)
##### Job Satisfaction: The employee's satisfaction with their job: (Very Low, Low, Medium, High)
##### Performance Rating: The employee's performance rating: (Low, Below Average, Average, High)
##### Number of Promotions: The total number of promotions the employee has received.
##### Distance from Home: The distance between the employee's home and workplace, in miles.
##### Education Level: The highest education level attained by the employee: (High School, Associate Degree, Bachelor’s Degree, Master’s Degree, PhD)
##### Marital Status: The marital status of the employee: (Divorced, Married, Single)
##### Job Level: The job level of the employee: (Entry, Mid, Senior)
##### Company Size: The size of the company the employee works for: (Small,Medium,Large)
##### Company Tenure: The total number of years the employee has been working in the industry.
##### Remote Work: Whether the employee works remotely: (Yes or No)
##### Leadership Opportunities: Whether the employee has leadership opportunities: (Yes or No)
##### Innovation Opportunities: Whether the employee has opportunities for innovation: (Yes or No)
##### Company Reputation: The employee's perception of the company's reputation: (Very Poor, Poor,Good, Excellent)
##### Employee Recognition: The level of recognition the employee receives:(Very Low, Low, Medium, High)

##### Attrition: Whether the employee has left the company, encoded as 0 (stayed) and 1 (Left).

### 3. Employee Attrition Prediction Dataset
The Employee Attrition Prediction Dataset contains data for 10,000 employees, designed for predictive modeling and analysis of employee attrition. It includes a variety of demographic, job-related, and performance metrics to help understand the factors contributing to employee turnover.

Key Features:

##### Employee_ID: Unique identifier for each employee.
##### Age: Age of the employee.
##### Gender: Gender of the employee.
##### Marital_Status: Marital status of the employee (Single, Married, Divorced).
##### Department: Department the employee works in (e.g., HR, IT, Sales, Marketing).
##### Job_Role: Specific role within the department (e.g., Manager, Analyst).
##### Job_Level: Level in the organizational hierarchy.
##### Monthly_Income: Monthly salary of the employee.
##### Hourly_Rate: Rate per hour for hourly employees.
##### Years_at_Company: Number of years the employee has been with the company.
##### Years_in_Current_Role: Number of years the employee has been in their current role.
##### Years_Since_Last_Promotion: Time since the employee’s last promotion.
##### Work_Life_Balance: Rating of work-life balance.
##### Job_Satisfaction: Rating of job satisfaction (1-5 scale).
##### Performance_Rating: Performance rating (1-5 scale).
##### Training_Hours_Last_Year: Number of training hours completed in the past year.
##### Overtime: Whether the employee works overtime (Yes/No).
##### Project_Count: Number of projects managed by the employee.
##### Average_Hours_Worked_Per_Week: Average working hours per week.
##### Absenteeism: Number of days the employee was absent in the past year.
##### Work_Environment_Satisfaction: Rating of work environment satisfaction.
##### Relationship_with_Manager: Rating of the relationship with the manager.
##### Job_Involvement: Rating of job involvement.
##### Distance_From_Home: Distance from home to the workplace (in kilometers).
##### Number_of_Companies_Worked: Total number of companies the employee has worked for.
##### Attrition: The target column (Yes/No) indicating whether the employee left the company.

### 4. HR Employee Attrition
#### Description:

This dataset provides comprehensive information on employees within an organization, including their demographics, job-related factors, and attrition status. It is valuable for understanding factors contributing to employee turnover and for developing predictive models to anticipate attrition within a workforce.

#### Features:

##### Age: The age of the employee.
##### Attrition: The attrition status of the employee (binary: yes/no).
##### BusinessTravel: Frequency of business travel (e.g., non-travel, travel frequently, etc.).
##### DailyRate: The daily rate of pay for the employee.
##### Department: The department in which the employee works.
##### DistanceFromHome: The distance from the employee's home to the workplace.
##### Education: The level of education attained by the employee.
##### EducationField: The field of education of the employee.
##### EmployeeCount: The count of employees (constant value for this dataset).
##### EmployeeNumber: A unique identifier for each employee.
##### EnvironmentSatisfaction: The level of satisfaction with the work environment.
##### Gender: The gender of the employee.
##### HourlyRate: The hourly rate of pay for the employee.
##### JobInvolvement: The level of involvement in job-related activities.
##### JobLevel: The level of the employee's job.
##### JobRole: The role or position of the employee.
##### JobSatisfaction: The level of job satisfaction reported by the employee.
##### MaritalStatus: The marital status of the employee.
##### MonthlyIncome: The monthly income of the employee.
##### MonthlyRate: The monthly rate of pay for the employee.
##### NumCompaniesWorked: The number of companies the employee has worked for.
##### Over18: Indicates if the employee is over 18 years old (binary: yes/no).
##### OverTime: Indicates if the employee works overtime (binary: yes/no).
##### PercentSalaryHike: The percentage increase in salary for the employee.
##### PerformanceRating: The performance rating of the employee.
##### RelationshipSatisfaction: The level of satisfaction with relationships at work.
##### StandardHours: The standard number of hours worked per week (constant value for this dataset).
##### StockOptionLevel: The level of stock options awarded to the employee.
##### TotalWorkingYears: The total number of years the employee has worked.
##### TrainingTimesLastYear: The number of training sessions attended by the employee last year.
##### WorkLifeBalance: The level of balance between work and personal life reported by the employee.
##### YearsAtCompany: The number of years the employee has worked at the current company.
##### YearsInCurrentRole: The number of years the employee has been in the current role.
##### YearsSinceLastPromotion: The number of years since the employee's last promotion.
##### YearsWithCurrManager: The number of years the employee has been with the current manager.

#### Usage:

This dataset can be used to analyze factors contributing to employee attrition and to develop predictive models for identifying employees at risk of leaving the organization.
It can also be used for exploring correlations between various employee attributes and job satisfaction, performance, and tenure.


## Технический анализ датасетов

In [1]:
import pandas as pd

In [2]:
df = pd.read_csv('train.csv')
print("\ntrain.csv:")
display(df.head(5))
print("\nSize of Dataset:", df.shape)
print("\nDataset info:")
df.info()


train.csv:


,Employee ID,Age,Gender,Years at Company,Job Role,Monthly Income,Work-Life Balance,Job Satisfaction,Performance Rating,Number of Promotions,...,Number of Dependents,Job Level,Company Size,Company Tenure,Remote Work,Leadership Opportunities,Innovation Opportunities,Company Reputation,Employee Recognition,Attrition
0,8410,31,Male,19,Education,5390,Excellent,Medium,Average,2,...,0,Mid,Medium,89,No,No,No,Excellent,Medium,Stayed
1,64756,59,Female,4,Media,5534,Poor,High,Low,3,...,3,Mid,Medium,21,No,No,No,Fair,Low,Stayed
2,30257,24,Female,10,Healthcare,8159,Good,High,Low,0,...,3,Mid,Medium,74,No,No,No,Poor,Low,Stayed
3,65791,36,Female,7,Education,3989,Good,High,High,1,...,2,Mid,Small,50,Yes,No,No,Good,Medium,Stayed
4,65026,56,Male,41,Education,4821,Fair,Very High,Average,0,...,0,Senior,Medium,68,No,No,No,Fair,Medium,Stayed



Size of Dataset: (59598, 24)

Dataset info:
<class 'pandas.DataFrame'>
RangeIndex: 59598 entries, 0 to 59597
Data columns (total 24 columns):
 #   Column                    Non-Null Count  Dtype
---  ------                    --------------  -----
 0   Employee ID               59598 non-null  int64
 1   Age                       59598 non-null  int64
 2   Gender                    59598 non-null  str  
 3   Years at Company          59598 non-null  int64
 4   Job Role                  59598 non-null  str  
 5   Monthly Income            59598 non-null  int64
 6   Work-Life Balance         59598 non-null  str  
 7   Job Satisfaction          59598 non-null  str  
 8   Performance Rating        59598 non-null  str  
 9   Number of Promotions      59598 non-null  int64
 10  Overtime                  59598 non-null  str  
 11  Distance from Home        59598 non-null  int64
 12  Education Level           59598 non-null  str  
 13  Marital Status            59598 non-null  str  
 14  Numb

## 1. Employee Attrition Classification Dataset
**Общая информация**

- **train.csv**: `(59598, 24)`
- **test.csv**: `(14900, 24)`
- **Всего записей**: **74 498**
- **Количество признаков**: 24
- **Типы данных**: 8 числовых + 16 категориальных
- **Пропуски**: **Отсутствуют**
- **Целевая переменная**: `Attrition` (`Stayed` / `Left`)

**Ключевые колонки для оценки выгорания** ✅

- `Work-Life Balance`
- `Job Satisfaction`
- `Overtime`
- `Leadership Opportunities`
- `Innovation Opportunities`
- `Company Reputation`
- `Employee Recognition`


In [3]:
df = pd.read_csv('employee_attrition_dataset_10000.csv')
print("\nemployee_attrition_dataset_10000.csv:")
display(df.head(5))
print("\nSize of Dataset:", df.shape)
print("\nDataset info:")
df.info()


employee_attrition_dataset_10000.csv:


,Employee_ID,Age,Gender,Marital_Status,Department,Job_Role,Job_Level,Monthly_Income,Hourly_Rate,Years_at_Company,...,Overtime,Project_Count,Average_Hours_Worked_Per_Week,Absenteeism,Work_Environment_Satisfaction,Relationship_with_Manager,Job_Involvement,Distance_From_Home,Number_of_Companies_Worked,Attrition
0,1,58,Male,Single,Finance,Manager,5,7332,81,24,...,No,9,48,16,4,1,1,49,3,No
1,2,48,Female,Divorced,HR,Assistant,4,6069,55,18,...,Yes,9,57,10,4,1,1,25,1,No
2,3,34,Female,Married,Marketing,Manager,4,11485,65,6,...,Yes,3,55,1,1,4,3,21,1,Yes
3,4,27,Female,Divorced,HR,Manager,4,18707,28,12,...,No,9,53,2,3,4,1,46,2,No
4,5,40,Male,Married,HR,Analyst,1,16398,92,3,...,No,1,54,11,1,1,1,43,4,No



Size of Dataset: (10000, 26)

Dataset info:
<class 'pandas.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 26 columns):
 #   Column                         Non-Null Count  Dtype
---  ------                         --------------  -----
 0   Employee_ID                    10000 non-null  int64
 1   Age                            10000 non-null  int64
 2   Gender                         10000 non-null  str  
 3   Marital_Status                 10000 non-null  str  
 4   Department                     10000 non-null  str  
 5   Job_Role                       10000 non-null  str  
 6   Job_Level                      10000 non-null  int64
 7   Monthly_Income                 10000 non-null  int64
 8   Hourly_Rate                    10000 non-null  int64
 9   Years_at_Company               10000 non-null  int64
 10  Years_in_Current_Role          10000 non-null  int64
 11  Years_Since_Last_Promotion     10000 non-null  int64
 12  Work_Life_Balance              10000 non-

## 2. Employee Attrition Prediction Dataset

**Общая информация**

- **Размер**: `(10000, 26)`
- **Типы данных**: 20 числовых + 6 категориальных
- **Пропуски**: **Отсутствуют**
- **Целевая переменная**: `Attrition` (`Yes` / `No`)

**Ключевые колонки для burnout scoring** ✅

- `Average_Hours_Worked_Per_Week`
- `Project_Count`
- `Absenteeism`
- `Work_Life_Balance`
- `Job_Satisfaction`
- `Work_Environment_Satisfaction`

In [4]:
df = pd.read_csv('WA_Fn-UseC_-HR-Employee-Attrition.csv')
print("\nIBM HR Analytics Employee Attrition & Performance dataset:")
display(df.head(5))
print("\nSize of Dataset:", df.shape)
print("\nDataset info:")
df.info()


IBM HR Analytics Employee Attrition & Performance dataset:


,Age,Attrition,BusinessTravel,DailyRate,Department,DistanceFromHome,Education,EducationField,EmployeeCount,EmployeeNumber,...,RelationshipSatisfaction,StandardHours,StockOptionLevel,TotalWorkingYears,TrainingTimesLastYear,WorkLifeBalance,YearsAtCompany,YearsInCurrentRole,YearsSinceLastPromotion,YearsWithCurrManager
0,41,Yes,Travel_Rarely,1102,Sales,1,2,Life Sciences,1,1,...,1,80,0,8,0,1,6,4,0,5
1,49,No,Travel_Frequently,279,Research & Development,8,1,Life Sciences,1,2,...,4,80,1,10,3,3,10,7,1,7
2,37,Yes,Travel_Rarely,1373,Research & Development,2,2,Other,1,4,...,2,80,0,7,3,3,0,0,0,0
3,33,No,Travel_Frequently,1392,Research & Development,3,4,Life Sciences,1,5,...,3,80,0,8,3,3,8,7,3,0
4,27,No,Travel_Rarely,591,Research & Development,2,1,Medical,1,7,...,4,80,1,6,3,3,2,2,2,2



Size of Dataset: (1470, 35)

Dataset info:
<class 'pandas.DataFrame'>
RangeIndex: 1470 entries, 0 to 1469
Data columns (total 35 columns):
 #   Column                    Non-Null Count  Dtype
---  ------                    --------------  -----
 0   Age                       1470 non-null   int64
 1   Attrition                 1470 non-null   str  
 2   BusinessTravel            1470 non-null   str  
 3   DailyRate                 1470 non-null   int64
 4   Department                1470 non-null   str  
 5   DistanceFromHome          1470 non-null   int64
 6   Education                 1470 non-null   int64
 7   EducationField            1470 non-null   str  
 8   EmployeeCount             1470 non-null   int64
 9   EmployeeNumber            1470 non-null   int64
 10  EnvironmentSatisfaction   1470 non-null   int64
 11  Gender                    1470 non-null   str  
 12  HourlyRate                1470 non-null   int64
 13  JobInvolvement            1470 non-null   int64
 14  JobLeve

## 3. IBM HR Analytics Employee Attrition & Performance

**Общая информация**

- **Размер**: `(1470, 35)`
- **Типы данных**: 26 числовых + 9 категориальных
- **Пропуски**: **Отсутствуют**
- **Целевая переменная**: `Attrition` (`Yes` / `No`)

In [5]:
df = pd.read_csv('employee_attrition_dataset.csv')
print("\nemployee_attrition_dataset.csv:")
display(df.head(5))
print("\nSize of Dataset:", df.shape)
print("\nDataset info:")
df.info()


employee_attrition_dataset.csv:


,Employee_ID,Age,Gender,Marital_Status,Department,Job_Role,Job_Level,Monthly_Income,Hourly_Rate,Years_at_Company,...,Overtime,Project_Count,Average_Hours_Worked_Per_Week,Absenteeism,Work_Environment_Satisfaction,Relationship_with_Manager,Job_Involvement,Distance_From_Home,Number_of_Companies_Worked,Attrition
0,1,58,Female,Married,IT,Manager,1,15488,28,15,...,No,6,54,17,4,4,4,20,3,No
1,2,48,Female,Married,Sales,Assistant,5,13079,28,6,...,Yes,2,45,1,4,1,2,25,2,No
2,3,34,Male,Married,Marketing,Assistant,1,13744,24,24,...,Yes,6,34,2,3,4,4,45,3,No
3,4,27,Female,Divorced,Marketing,Manager,1,6809,26,10,...,No,9,48,18,2,3,1,35,3,No
4,5,40,Male,Divorced,Marketing,Executive,1,10206,52,29,...,No,3,33,0,4,1,3,44,3,No



Size of Dataset: (1000, 26)

Dataset info:
<class 'pandas.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 26 columns):
 #   Column                         Non-Null Count  Dtype
---  ------                         --------------  -----
 0   Employee_ID                    1000 non-null   int64
 1   Age                            1000 non-null   int64
 2   Gender                         1000 non-null   str  
 3   Marital_Status                 1000 non-null   str  
 4   Department                     1000 non-null   str  
 5   Job_Role                       1000 non-null   str  
 6   Job_Level                      1000 non-null   int64
 7   Monthly_Income                 1000 non-null   int64
 8   Hourly_Rate                    1000 non-null   int64
 9   Years_at_Company               1000 non-null   int64
 10  Years_in_Current_Role          1000 non-null   int64
 11  Years_Since_Last_Promotion     1000 non-null   int64
 12  Work_Life_Balance              1000 non-null

## 4. HR Employee Attrition

**Общая информация**

- **Размер**: `(1000, 26)`
- **Типы данных**: 20 числовых + 6 категориальных
- **Пропуски**: **Отсутствуют**
- **Целевая переменная**: `Attrition` (`Yes` / `No`)

# 📊 Выводы
## 1. Employee Attrition Classification Dataset

**Общая информация**
- **train.csv**: 59 598 записей
- **test.csv**: 14 900 записей
- **Всего**: 74 498 записей
- Количество колонок: 24
- Пропуски: отсутствуют
- Целевая переменная: `Attrition` (`Stayed` / `Left`)

**Ключевые преимущества для burnout scoring** ✅
- Прямые индикаторы выгорания: `Work-Life Balance`, `Job Satisfaction`, `Overtime`
- Дополнительные важные признаки: `Leadership Opportunities`, `Innovation Opportunities`, `Employee Recognition`, `Company Reputation`
- Хорошее разнообразие должностей (Education, Healthcare, Technology, Finance, Media)
- Хорошо сбалансированные классы (~52% Stayed / 48% Left)

**Сильные стороны**
- Самый большой объём данных
- Высокое качество и разнообразие признаков
- Уже разделён на train/test

**Слабые стороны**
- Отсутствуют количественные метрики нагрузки (среднее количество часов, проекты)

**Общая оценка**: ⭐⭐⭐⭐⭐ **Лучший выбор для основного проекта**

---

## 2. Employee Attrition Prediction Dataset

**Общая информация**
- Записей: 10 000
- Колонок: 26
- Пропуски: отсутствуют
- Целевая переменная: `Attrition` (`Yes` / `No`)

**Ключевые преимущества для burnout scoring** ✅
- Отличные количественные показатели: `Average_Hours_Worked_Per_Week`, `Project_Count`, `Absenteeism`, `Training_Hours_Last_Year`
- Хорошие качественные признаки: `Work_Life_Balance`, `Job_Satisfaction`, `Work_Environment_Satisfaction`

**Сильные стороны**
- Прямые метрики рабочей перегрузки
- Удобный размер для экспериментов

**Слабые стороны**
- Сильный дисбаланс классов (~16% Yes)

**Общая оценка**: ⭐⭐⭐⭐ **Отличный дополнительный датасет**

---

## 3. IBM HR Analytics Employee Attrition & Performance

**Общая информация**
- Записей: 1 470
- Колонок: 35
- Пропуски: отсутствуют
- Целевая переменная: `Attrition` (`Yes` / `No`)

**Ключевые особенности**
- Классический учебный датасет
- Содержит: `OverTime`, `WorkLifeBalance`, `JobSatisfaction`, `EnvironmentSatisfaction`, `RelationshipSatisfaction`

**Сильные стороны**
- Хорошо изучен сообществом
- Много интерпретируемых признаков
- Удобен для бенчмарков

**Слабые стороны**
- Маленький размер данных

**Общая оценка**: ⭐⭐⭐ **Подходит для прототипирования**

---

## 4. HR Employee Attrition

**Общая информация**
- Записей: 1 000
- Колонок: 26
- Пропуски: отсутствуют
- Целевая переменная: `Attrition` (`Yes` / `No`)

**Ключевые особенности**
- Мини-версия датасета на 10 000 записей
- Содержит те же признаки нагрузки (часы работы, прогулы, проекты)

**Сильные стороны**
- Компактный, удобен для быстрых тестов

**Слабые стороны**
- Слишком маленький объём для надёжного моделирования

**Общая оценка**: ⭐⭐ **Только для начальных экспериментов**

---

## 📈 Сравнительная таблица

| Критерий                      |    Employee Attrition Classification Dataset    | Employee Attrition Prediction Dataset     | IBM HR Analytics Employee Attrition & Performance     | HR Employee Attrition     |
|-------------------------------|------------------|------------------|-----------------|----------------|
| Размер данных                 | ★★★★★           | ★★★★            | ★★             | ★             |
| Баланс классов                | ★★★★★           | ★★              | ★★             | ★★            |
| Релевантность для burnout     | ★★★★★           | ★★★★★           | ★★★★           | ★★★★★         |
| Разнообразие признаков        | ★★★★★           | ★★★★            | ★★★            | ★★★           |
| **Рекомендация**              | **Основной**     | **Дополнительный** | Для сравнения  | Для тестов    |

---

### 🎯 Итоговая рекомендация

**Основной датасет:** `Employee Attrition Classification Dataset` 

**Дополнительный датасет:** `Employee Attrition Prediction Dataset`

Такое сочетание позволит создать объёмную и качественную модель оценки риска выгорания.